test/unit/puppet_fact_parser_test.rb

```diff
@@ -510,6 +510,15 @@ def setup
             assert_kind_of expected, subject.ram
           end
 
+          test "#kernel_version" do
+            if facterversion.to_f >= 3.14
+              assert_equal '3.10.0-957.12.2.el7.x86_64', subject.kernel_version
+            else
+              assert_equal '3.10.0-123.el7.i386', subject.kernel_version
+            end
+            assert_kind_of String, subject.kernel_version
```

Comment:
This feels redundant if you already have an exact match above. There you already know it's a string. The logic in most of these tests is that we don't care about the exact value (since they may change are are largely irrelevant). That's why it checks on the data type so that it at least retrieves something.

In this case I'd only keep `assert_kind_of String, subject.kernel_version` and remove the exact matches. Mostly because someone might record newer facts with a new kernel (note the data comes from https://github.com/voxpupuli/facterdb). That could easily break the tests, which is annoying for all developers. You could have some regular expression, but I don't think it gives a higher guarantee about safety.

Reply:
@ekohl Updated! Thanks for the detailed explanation around the tests, I was struggling with what to test and tried to find the best fit. 

I have updated the pr based on your feedback and reading what you wrote I can see why the other unit tests were just testing the data type being returned. This is my first big Foreman PR so thank you for taking the time to review and give me guidance. Marek wanted me to add a few more facts (in a separate pr) so this guidance will help me with those as well.